common/workunit/workunit.cpp

```diff
@@ -3004,6 +3005,7 @@ class CDaliWorkUnitFactory : public CWorkUnitFactory, implements IDaliClientShut
                 conn->reload();
             }
         }
+        waiter->unsubscribe();
```

Comment:
It's not needed (it was also in the 
```
            case WUStateWait:
                if(returnOnWaitState)
```

.. block, but it's called by the WorkUnitWaiter destructor when this method goes out of scope anyway.
For same reason it was taken out in this method, should really have been taken out of waitForWorkUnitAction() too.